From: Christian Marillat <marillat@debian.org>
Date: Mon, 28 Jul 2014 10:35:04 +0200
Subject: 03_File.pm

---
 lib/RRD/File.pm |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/RRD/File.pm b/lib/RRD/File.pm
index cd23534..2b3a0f5 100644
--- a/lib/RRD/File.pm
+++ b/lib/RRD/File.pm
@@ -196,7 +196,7 @@ sub loadHeader {
     {
         my ($block);
         my ($last_up, $last_up_usec);
-        if ($v eq "0001" || $v eq "0002") {
+        if ($v eq "0001" || $v eq "0002" || $v eq "0003") {
            $block = $self->_readNextBlock(sizeof($fmt->format('liveHead')));
            croak("Could not read live header") unless (defined($block));
            $last_up = unpack($fmt->format('liveHead'), $block);
