From: Christian Marillat <marillat@debian.org>
Date: Mon, 28 Jul 2014 10:35:04 +0200
Subject: 05_systemperf

---
 sample-config/systemperf/Defaults |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/sample-config/systemperf/Defaults b/sample-config/systemperf/Defaults
index 856107d..8b50061 100644
--- a/sample-config/systemperf/Defaults
+++ b/sample-config/systemperf/Defaults
@@ -29,7 +29,7 @@ target --default--
     max-size      = undef
 
 OID    hrSystemNumUsers     1.3.6.1.2.1.25.1.5.0
-OID    hrSystemProcesses    1.3.6.1.2.1.25.1.6
+OID    hrSystemProcesses    1.3.6.1.2.1.25.1.6.0
 OID    hrStorageName        1.3.6.1.2.1.25.2.3.1.3
 OID    hrStorageSize        1.3.6.1.2.1.25.2.3.1.5
 OID    hrStorageUsed        1.3.6.1.2.1.25.2.3.1.6
