From: Christian Marillat <marillat@debian.org>
Date: Mon, 28 Jul 2014 10:35:04 +0200
Subject: 06_Format.pm

===================================================================
---
 lib/RRD/Format.pm |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/lib/RRD/Format.pm b/lib/RRD/Format.pm
index 55fcb72..52dcb19 100644
--- a/lib/RRD/Format.pm
+++ b/lib/RRD/Format.pm
@@ -123,6 +123,7 @@ sub setArch {
         $self->{'cdpDef'} = "d L x4 x64";
 
         $self->{'liveHead'} = "L";
+        $self->{'liveHead3'} = "L L";
         $self->{'rraPtr'} = "L";
         $self->{'element'} = "d";
 
@@ -150,7 +151,7 @@ sub setArch {
         $self->{'liveHead'} = "L";
         $self->{'rraPtr'} = "L";
         $self->{'element'} = "d";
-    } elsif ( $archname eq 'alpha-linux' ) {
+    } elsif ( $archname eq 'alpha-linux' or $archname eq 'alpha-linux-gnu') {
         $self->{'statHead'} = "a4 a5 x7 d Q Q Q x80";
         $self->{'dsDef'} = "a20 a20 Q d d x56";
         $self->{'rraDef'} = "a20 x4 Q Q d x72";
