From: Christian Marillat <marillat@debian.org>
Date: Mon, 28 Jul 2014 10:35:04 +0200
Subject: 07_Format.pm.x86_64

===================================================================
---
 lib/RRD/Format.pm |   10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/lib/RRD/Format.pm b/lib/RRD/Format.pm
index 52dcb19..363e50a 100644
--- a/lib/RRD/Format.pm
+++ b/lib/RRD/Format.pm
@@ -160,6 +160,16 @@ sub setArch {
         $self->{'liveHead'} = "Q";
         $self->{'rraPtr'} = "Q";
         $self->{'element'} = "d";
+    } elsif ( $archname =~ /^x86_64-linux/ ) {
+        $self->{'statHead'} = "a4 a5 x7 d Q Q Q x80";
+        $self->{'dsDef'} = "a20 a20 Q d d x56";
+        $self->{'rraDef'} = "a20 x4 Q Q d x72";
+        $self->{'liveHead'} = "Q Q";
+        $self->{'liveHead3'} = "Q Q";
+        $self->{'pdpDef'} = "a30 x2 Q d x64";
+        $self->{'cdpDef'} = "d Q x64";
+        $self->{'rraPtr'} = "Q";
+        $self->{'element'} = "d";
     } elsif ( $archname eq 'sparc64-netbsd' ) {
         $self->{'statHead'} = "a4 a5 x7 d Q Q Q x80";
         $self->{'dsDef'} = "a20 a20 Q d d x56";
